modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java

```diff
@@ -287,7 +294,11 @@ public static Object fromInternal(ExecutionContext<?> ectx, Object val, Type sto
             return Period.of((Integer) val / 12, (Integer) val % 12, 0);
         } else if (storageType == byte[].class && val instanceof ByteString) {
             return ((ByteString) val).getBytes();
+        } else if (storageType == UUID.class) {
```

Comment:
do we need it just for assertion check and if so - is assertion is enough here?

Reply:
It is not sufficient because else branch is called for all other type/value combinations. 
And if guard of UUID tests that a value of UUID data type is UUID and nothing else.